packages/embedder/indexing/retrospectiveDiscussionTopic.ts

```diff
@@ -0,0 +1,267 @@
+import {Selectable} from 'kysely'
+import prettier from 'prettier'
+
+import getRethink, {RethinkSchema} from 'parabol-server/database/rethinkDriver'
+import {DataLoaderWorker} from 'parabol-server/graphql/graphql'
+import getKysely from 'parabol-server/postgres/getKysely'
+import {DB} from 'parabol-server/postgres/pg'
+
+import Comment from 'parabol-server/database/types/Comment'
+import DiscussStage from 'parabol-server/database/types/DiscussStage'
+import MeetingRetrospective, {
+  isMeetingRetrospective
+} from 'parabol-server/database/types/MeetingRetrospective'
+import {RDatum} from 'parabol-server/database/stricterR'
+
+import generateHourIntervals from './generateHourIntervals'
+import {DBInsert} from '../embedder'
+import {upsertEmbeddingsMetaRows} from './embeddingsTablesOps'
+
+export interface EmbeddingsJobQueueRetrospectiveDiscussionTopic
+  extends Omit<DB['EmbeddingsJobQueue'], 'objectType'> {
+  objectType: 'retrospectiveDiscussionTopic'
+}
+
+// Here's a generic reprentation of the text generated here:
+
+// A topic "<DISCUSSION_TOPIC>" was discussed during the meeting "<MEETING_NAME>"
+// that followed the "<TEMPLATE_NAME>" template.
+//
+// <FOR EACH PROMPT>
+// Participants were prompted with, "<PROMPT>: <PROMPT DESCRIPTION>".
+//  - <PARTICIPANT|ANONYMOUS> wrote, "<RESPONSE>"
+//
+// </FOR EACH PROMPT>
+//
+// <IF DISCUSSION>
+// A discussion was held. <DISCUSSION SUMMARY>
+// </END IF DISCUSSION>
+
+const IGNORE_COMMENT_USER_IDS = ['parabolAIUser']
+
+const pg = getKysely()
+
+export async function refreshEmbeddingsMeta(dataLoader: DataLoaderWorker) {
+  const r = await getRethink()
+  const {createdAt: newestMeetingDate} = (await r
+    .table('NewMeeting')
+    .max({index: 'createdAt'})
+    .run()) as unknown as RethinkSchema['NewMeeting']['type']
+  const {createdAt: oldestMeetingDate} = (await r
+    .table('NewMeeting')
+    .min({index: 'createdAt'})
+    .run()) as unknown as RethinkSchema['NewMeeting']['type']
+
+  const {newestIndexDate} = (await pg
+    .selectFrom('EmbeddingsMetadata')
+    .select(pg.fn.max('refUpdatedAt').as('newestIndexDate'))
+    .executeTakeFirst()) ?? {newestIndexDate: null}
+  const startDateTime = newestIndexDate || oldestMeetingDate
+  console.log(`Index will consider adding items from ${startDateTime} to ${newestMeetingDate}`)
+
+  if (startDateTime.getTime() === newestMeetingDate.getTime()) return
+
+  for (const interval of generateHourIntervals(startDateTime, newestMeetingDate)) {
```

Comment:
+1 I think `generateHourIntervals` is being used to ensure a tolerable batch size. if that's correct, i don't think it's a good heuristic because meeting start times are not uniformly distributed. 

instead, in your query use `.orderBy('createdAt').limit(BATCH_SIZE)` and on each iteration of the loop set your `between` start time to the last time of the previous batch. be sure to support the case of 2 meetings starting at the exact same time!

Reply:
Oh, nevermind. `summarySentAt` has no index. `createdAt it is.